tests/TestCase/Routing/RouterTest.php

```diff
@@ -2091,6 +2092,14 @@ public function testRouteParamDefaults(): void
         }
     }
 
+    public function testRouteInvalidDefaults(): void
+    {
+        $this->expectException(ErrorException::class);
```

Comment:
Anyone knows how to tell phpunit that the error is expected? This isn't working.

Reply:
They are split into different functions now. expectError() I think.

Reply:
There should be separate functions now. expectError() I think.

Reply:
`expectError()` is gone and the only alternative is to resort to error handler juggling :(

Reply:
we could write more helper methods like `$this->deprecated()` to also assert other `trigger_error` calls.

Reply:
I don't see it in the documentation anymore either. Looks like phpunit only wants to alert you to use deprecated code not testing a framework with deprecations.